internal/cli/destroy.go

```diff
@@ -27,6 +27,21 @@ func (c *DestroyCommand) Run(args []string) int {
 	}
 
 	err := c.DoApp(c.Ctx, func(ctx context.Context, app *clientpkg.App) error {
+		if !c.flagForce {
+			// show confirmation dialog asking user for approval
+			choice, err := app.UI.Input(&terminal.Input{
+				Prompt: "Are you sure you want to destroy this app? Type 'yes' to confirm.",
+				Style:  terminal.WarningStyle,
+			})
+			if err != nil {
+				return err
+			}
+			if choice != "yes" {
+				app.UI.Output("Aborting destroy.")
+				return nil
+			}
+		}
```

Comment:
Since it looks like at the moment our App UI only implements noninteractive, this condition branch should instead return an error to the user that `auto-confirm` is required for destroying the app.

```suggestion
		if !c.confirm {
			app.UI.Output("Destroying app %q requires confirmation with `-auto-approve`.", app.Ref().GetApplication(), terminal.WithWarningStyle())
			return nil
		}
```